### ISSUE TITLE
Hanging when Matching Filenames in 0.12

### ISSUE THREAD

**Issue by fab-jul** [tensorflow/tensorflow#6190]
I have a network which uses `tf.train.shuffle_batch` with a `tf.train.string_input_produces`. The network worked in 0.11, but hangs in 0.12. 

Disclaimer: I don't have access to 0.11 anymore and I changed some other stuff, but I'm fairly sure that the important bit is still the same.

Hanging means: I don't get further than the "Creating TensorFlow device" message.

The reason I'm thinking it's due to the queue is because after enabling the `log_device_placement` flag, the last output I'm seeing is

    I tensorflow/core/common_runtime/simple_placer.cc:827] 
        matching_filenames/MatchingFiles/pattern

Also, using a barebowns graph like

```
a = tf.constant(4)
b = tf.constant(2)
s = tf.Session()
s.run(a + b)
```

works, so I don't think it is due to my TensorFlow installation.

The hang is reproducible for me. I don't know how to enable further logging but would be glad to do so.

---

The code to create the queue is as follows, basically it's the same as [in the tutorial](https://www.tensorflow.org/versions/r0.12/how_tos/reading_data/index.html)

```
filenames = tf.train.match_filenames_once(filenames_glob)
filename_queue = tf.train.string_input_producer(
    filenames, num_epochs=num_epochs, shuffle=shuffle)
image = _read_pngs(filename_queue)

min_after_dequeue = 100
capacity = min_after_dequeue + 1 * batch_size

return tf.train.shuffle_batch(
    [image],
    batch_size=batch_size, capacity=capacity,
    min_after_dequeue=min_after_dequeue,
    seed=666, name=name)
```
